### DISCUSSION TITLE
Auto close/fill option for vector brush.

### DISCUSSION THREAD

**Discussion by beeheemooth** [opentoonz/opentoonz#6798] [Roadmap]
Working with many vector groups is always associated with the routine of closing regions, grouping and filling.
Is it possible to make the vector brush mode in which the first and last points are connected and the drawn stroke forms a closed and filled with color area? The Polygonal Tool is very cumbersome to create such forms, it is very mechanical, non-artistic, lacks a free hand.

Options: auto close, auto group, auto fill.

![ezgif-4-384eb5f78f8d](https://user-images.githubusercontent.com/25724133/85223087-c3524000-b3c8-11ea-91c3-9b5c08340e43.gif)

**Comment by phillycheese007** [comment]
Yes please!

**Comment by artisteacher** [comment]
If snapping is on, and the stroke starts & ends at the same place, it already auto-closes. It's a little iffy if snapping is on low, but seems reliable on medium. So part of the functionality is already there.

I agree that polyline is too fussy for making certain shapes. It would be easier to work with though, if it could make cusp points (where the 2nd handle can be moved independently of the first to get 2 different curves on either side of a control point). Then it would have all the functionality of a pen tool. There's something to be said for having complete control over how many control points are created.

**Comment by melieconiek** [comment]
I'd be game for this, it'd make lasso style fills a lot easier.

**Comment by phillycheese007** [comment]
@beeheemooth 

Im confused....it looks like youa already have this function? How are you getting that automatic fill w/  brush?

**Comment by beeheemooth** [comment]
@phillycheese007 this is video editing

**Comment by beeheemooth** [comment]
Note; this feature was implemented today in Tahoma by @turtletooth.
 Thank you!
https://github.com/turtletooth/tahoma/pull/197

**Comment by RodneyBaker** [comment]
Thanks @beeheemooth 
We'll keep this feature request open here pending adding the feature to Opentoonz.

Tagging as 'roadmap'.
